proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlaySoundPacketTranslator.java

```diff
@@ -14,37 +14,47 @@
 
 import com.github.steveice10.mc.protocol.data.game.world.sound.BuiltinSound;
 import com.github.steveice10.mc.protocol.data.game.world.sound.CustomSound;
+
+import org.dragonet.proxy.DragonProxy;
 import org.dragonet.proxy.network.UpstreamSession;
 import org.dragonet.proxy.network.translator.IPCPacketTranslator;
 import com.github.steveice10.mc.protocol.packet.ingame.server.world.ServerPlaySoundPacket;
+
 import org.dragonet.common.maths.BlockPosition;
 import org.dragonet.protocol.PEPacket;
 import org.dragonet.protocol.packets.PlaySoundPacket;
 
 public class PCPlaySoundPacketTranslator implements IPCPacketTranslator<ServerPlaySoundPacket> {
 
-    public PEPacket[] translate(UpstreamSession session, ServerPlaySoundPacket packet) {
-        try {
-            String soundName;
-
-            if (BuiltinSound.class.isAssignableFrom(packet.getSound().getClass())) {
-                BuiltinSound sound = (BuiltinSound) packet.getSound();
-                soundName = sound.name();
-            } else {
-                soundName = ((CustomSound) packet.getSound()).getName();
-            }
-            if (soundName == null) {
-                return null;
-            }
-            PlaySoundPacket pk = new PlaySoundPacket();
-            pk.blockPosition = new BlockPosition((int) packet.getX(), (int) packet.getY(), (int) packet.getZ());
-            pk.name = soundName;
-            pk.volume = packet.getVolume();
-            pk.pitch = packet.getPitch();
-            return new PEPacket[]{pk};
-        } catch (Exception e) {
-            e.printStackTrace();
-            return null;
-        }
-    }
+	public PEPacket[] translate(UpstreamSession session, ServerPlaySoundPacket packet) {
+		try {
+			String soundName;
+			
+			if (BuiltinSound.class.isAssignableFrom(packet.getSound().getClass())) {
+				if(DragonProxy.getInstance().getSoundTranslator().isIgnored((BuiltinSound)packet.getSound())) {
+					return null;
+				}
+				if (DragonProxy.getInstance().getSoundTranslator().isTranslatable((BuiltinSound)packet.getSound())) {
```

Comment:
Why does everyone, even @HoverEpic, do this?

UpstreamSession has a getProxy() method.

Reply:
I'll change it myself im just saying for future reference.